### ISSUE TITLE
Logging runs on every pageload due to inadequate settings check

### ISSUE THREAD

**Issue by johnwatkins0** [PopupMaker/Popup-Maker#687]
PR forthcoming.

https://github.com/PopupMaker/Popup-Maker/blob/befbc4f45617db4eec9b23512b7e939415e1ee21/includes/importer/easy-modal-v2.php#L211-L222

As you can see above, this function checks whether `$popmake_options['enable_easy_modal_compatibility_mode']` is set but *does not* check that  it is empty or false. As a result, the conditional always evaluates to true (after the PUM settings have been saved for the first time). 

Then:

https://github.com/PopupMaker/Popup-Maker/blob/5e6189666ce37496301a276a65c4268c8075b69b/includes/deprecated-filters.php#L158-L171

One of the filters added in that conditional causes the conditional in the above function to also evaluate to true. So logging is attempted on every page load. On a site I work on, this operation adds significantly -- 30 - 60ms -- to every pageload. 

I have seen there is a constant to disable logging, but that constant is checked after the log file is retrieved and created if needed, which is big part of the processing time. For my case it makes more sense to fix the `isset` check and not have logging attempted at all.

**Comment by danieliser** [comment]
@johnwatkins0 - hmm, easy enough to correct. Change the user check to instead use `pum_get_option( 'enable_easy_modal_compatibility_mode' )`

That said how large is your log file? Our testing showed anything under 1mb would have little to no impact, which is why the log is self truncating at just under 1mb

**Comment by johnwatkins0** [comment]
Hi, @danieliser. Yes, the impact in my local environment is minimal, but the production site's uploads directory is not in the same server environment, so the operations relating to reading files (e.g., `file_exists`, `filesize`, etc.) take a bit longer, so there's a gotcha in that particular use case. We've had similar issues with other plugins using file-related PHP functions.

After pushing a workaround (not the PR I submitted -- we'll use that when it's merged) to filter that option out of the PUM settings, the time spent in the `pum_popup_data_attr` went from around 50ms per page load down to around 0.002s.

I updated the PR to use the function you suggested.
